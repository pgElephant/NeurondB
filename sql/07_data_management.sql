-- Test data management functions
CREATE TABLE test_vectors_dm (
    id serial PRIMARY KEY,
    embedding vector,
    created_at timestamptz DEFAULT now(),
    last_accessed timestamptz,
    is_compressed boolean DEFAULT false
);

INSERT INTO test_vectors_dm (embedding) VALUES 
    ('[1.0, 0.0, 0.0]'),
    ('[0.0, 1.0, 0.0]'),
    ('[0.0, 0.0, 1.0]');

-- Test vacuum (should return count)
SELECT vacuum_vectors('test_vectors_dm', false) >= 0 AS vacuum_executed;

-- Test compress (should return count)
SELECT compress_cold_tier('test_vectors_dm', 30) >= 0 AS compress_executed;

-- Test rebalance
SELECT rebalance_index('test_index', 0.8) AS rebalance_executed;

DROP TABLE test_vectors_dm;

