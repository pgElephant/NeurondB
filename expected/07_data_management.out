-- Test data management functions
CREATE TABLE test_vectors_dm (
    id serial PRIMARY KEY,
    embedding vector,
    created_at timestamptz DEFAULT now(),
    last_accessed timestamptz,
    is_compressed boolean DEFAULT false
);
ERROR:  type "vector" does not exist
LINE 3:     embedding vector,
                      ^
INSERT INTO test_vectors_dm (embedding) VALUES 
    ('[1.0, 0.0, 0.0]'),
    ('[0.0, 1.0, 0.0]'),
    ('[0.0, 0.0, 1.0]');
ERROR:  relation "test_vectors_dm" does not exist
LINE 1: INSERT INTO test_vectors_dm (embedding) VALUES 
                    ^
-- Test vacuum (should return count)
SELECT vacuum_vectors('test_vectors_dm', false) >= 0 AS vacuum_executed;
ERROR:  function vacuum_vectors(unknown, boolean) does not exist
LINE 1: SELECT vacuum_vectors('test_vectors_dm', false) >= 0 AS vacu...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
-- Test compress (should return count)
SELECT compress_cold_tier('test_vectors_dm', 30) >= 0 AS compress_executed;
ERROR:  function compress_cold_tier(unknown, integer) does not exist
LINE 1: SELECT compress_cold_tier('test_vectors_dm', 30) >= 0 AS com...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
-- Test rebalance
SELECT rebalance_index('test_index', 0.8) AS rebalance_executed;
ERROR:  function rebalance_index(unknown, numeric) does not exist
LINE 1: SELECT rebalance_index('test_index', 0.8) AS rebalance_execu...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
DROP TABLE test_vectors_dm;
ERROR:  table "test_vectors_dm" does not exist
